### ISSUE TITLE
Add an option to switch distributions order in the KLDivergence.

### ISSUE THREAD

**Issue by i-aki-y** [Lightning-AI/torchmetrics#2369]
## 🚀 Feature

I want to add an option to switch the order of distributions in the KLDivergence.

Ex.

```py
kl_divergence(P, Q, order="pq") := P log (P / Q)  # default
kl_divergence(P, Q, order="qp") := Q log (Q / P)
```
### Motivation

The current kl_divergence is defined as follows:

```py
kl_divergence(P, Q) := P log (P / Q)
```

This is inconsistent with other metrics regarding the order of arguments.

The documentation says that "P usually represents a distribution over data."
In other words, the P represents the target.

However, the first argument of the current kl_divergence is P (= target) and is inconsistent with other metrics expecting prediction as the first argument.

For example, the PyTorch KLDivLoss is consistent with the conventional interpretation, but the argument order is opposite of the tochmetrics version.

```py
# pytorch KLDivLoss
kl_divergence(y_pred, y_true) = y_true log (y_true / y_pred)

# https://pytorch.org/docs/stable/generated/torch.nn.KLDivLoss.html#torch.nn.KLDivLoss
```

This is confusing. 

There is a practical issue that makes it difficult to use KLDivergence in the MetricCollection because MetricCollection expects all metrics to accept arguments in the same order: (preds, target).

### Pitch

I want to enable to accept the argument (predictions, targets) in KLDivergence and use it in MetricCollection

Ex.

```py
metrics = MetricCollection([
    KLDivergence(order="qp"),
    MeanAbsoluteError(),
    ...
])
```

### Alternatives

Redefine to switch the argument meanings:

```
kl_divergence(Q, P) = P log (P / Q)
```

But this breaks backward compatibility.

### Additional context

no

**Comment by SkafteNicki** [comment]
@Borda opinions here? I think the better option is actually to switch the arguments for the metric to be consistent with the rest of torchmetrics and rename to `preds`, `target`. We then need to deprecate correctly.
